osu.Game/Overlays/BeatmapSet/Scores/ScoresContainer.cs

```diff
@@ -44,100 +39,165 @@ public ScoresContainer()
+                                        RelativeSizeAxes = Axes.X,
+                                        AutoSizeAxes = Axes.Y,
+                                        Direction = FillDirection.Vertical,
+                                        Spacing = new Vector2(0, 5),
+                                    },
+                                    scoreTable = new ScoreTable
+                                    {
+                                        Anchor = Anchor.TopCentre,
+                                        Origin = Anchor.TopCentre,
+                                    }
+                                },
+                            }
+                        },
                     }
-                },
-                loadingAnimation = new LoadingAnimation
-                {
-                    Alpha = 0,
-                    Margin = new MarginPadding(20)
-                },
+                }
             };
+
+            Loading = true;
         }
 
         [BackgroundDependencyLoader]
         private void load(OsuColour colours)
         {
             background.Colour = colours.Gray2;
-            updateDisplay();
         }
 
-        private bool loading
+        private bool loading;
+
+        public bool Loading
         {
-            set => loadingAnimation.FadeTo(value ? 1 : 0, fade_duration);
+            get => loading;
+            set
+            {
+                loading = value;
+
+                if (value)
+                {
+                    loadingContainer.Show();
+                    contentContainer.Hide();
+                }
+                else
+                {
+                    loadingContainer.Hide();
+
+                    if (scores == null || scores?.Scores.Count < 1)
+                        contentContainer.Hide();
+                    else
+                        contentContainer.Show();
+                }
+            }
         }
 
-        private GetScoresRequest getScoresRequest;
-        private IReadOnlyList<ScoreInfo> scores;
+        private APILegacyScores scores;
 
-        public IReadOnlyList<ScoreInfo> Scores
+        public APILegacyScores Scores
         {
             get => scores;
             set
             {
-                getScoresRequest?.Cancel();
                 scores = value;
 
                 updateDisplay();
             }
         }
 
-        private BeatmapInfo beatmap;
-
-        public BeatmapInfo Beatmap
+        private void updateDisplay()
         {
-            get => beatmap;
-            set
-            {
-                beatmap = value;
+            topScoresContainer.Clear();
 
-                Scores = null;
+            scoreTable.Scores = scores?.Scores.Count > 1 ? scores.Scores : new List<APILegacyScoreInfo>();
+            scoreTable.FadeTo(scores?.Scores.Count > 1 ? 1 : 0);
 
-                if (beatmap?.OnlineBeatmapID.HasValue != true)
-                    return;
+            if (scores?.Scores.Any() == true)
```

Comment:
You can remove "== true" part

Reply:
boolean == const boolean is redundant

Reply:
we are using a null check here, so it's required

Reply:
`scores?.Scores.Any() ?? false`